lib/manager/terraform/lockfile/hash.ts

```diff
@@ -114,19 +101,11 @@ export async function calculateHashes(
 }
 
 export async function createHashes(
+  registryURL: string,
   repository: string,
   version: string
 ): Promise<string[]> {
-  // check cache for hashes
-  const repositoryRegexResult = repositoryRegex.exec(repository);
-  if (!repositoryRegexResult) {
-    // non hashicorp builds are not supported at the moment
-    return null;
-  }
-  const lookupName = repositoryRegexResult.groups.lookupName;
-  const backendLookUpName = `terraform-provider-${lookupName}`;
-
-  const cacheKey = `${TerraformProviderDatasource.defaultRegistryUrls[1]}/${repository}/${lookupName}-${version}`;
+  const cacheKey = `${registryURL}/${repository}-${version}`;
```

Comment:
```suggestion
  const cacheKey = `${registryURL}/${repository}/${version}`;
```

Unify, in datasource you are using a slash too.

@JamieMagee Thinking if we can use cache decorator here too?

Reply:
Yeah, it looks like a good candidate. You can use the cache decorator on any async method.